### PR TITLE
Solve rocket frame issue clean up opt traj generator

### PR DESCRIPTION
**Pull request by TunVunDerBunn**
For the rocket_frame issue, there is also a trello card.

### CODE REVIEW

File: src/eulersimcon_backend/components/optimal_trajectory/optimal_trajectory_generator.py [hunk 1]
```diff
@@ -263,58 +213,40 @@ def run(self, raw_options) -> OptimalTrajectoryModel:
...
 
-    # function to extract the trajectory and apogee (I dont know if there is a better way to do this)
-    def extract_trajectory_and_apogee(self, raw_options):
-        # traj = np.zeros(len(self.simulation.flight_steps))
-        # for i in range(len(traj)):
-        #     traj[i] = self.simulation.flight_steps[i].rocket_state.X_z
-        # apogee = np.max(traj)
-
+    # function to extract the trajectory and apogee
+    def get_apogee(self, raw_options):
```
Comment (mstoelzle) [inline src/eulersimcon_backend/components/optimal_trajectory/optimal_trajectory_generator.py:217]: Why does it still need the raw options?

Comment (TunVunDerBunn) [reply src/eulersimcon_backend/components/optimal_trajectory/optimal_trajectory_generator.py:217]: raw_options are needed for the init of the flight_analyzer

Comment (mstoelzle) [reply src/eulersimcon_backend/components/optimal_trajectory/optimal_trajectory_generator.py:217]: Yes, but why does the flight analyzer need the raw options? Are there still dependencies?

Comment (TunVunDerBunn) [reply src/eulersimcon_backend/components/optimal_trajectory/optimal_trajectory_generator.py:217]: yes we need it to init the environment, that if rocket frame is active, we can compute the mach number

**Pull request merged**